### DISCUSSION TITLE
How to selectively skip an object member when serializing?

### DISCUSSION THREAD

**Discussion by mputic365** [spec-first/connexion#1904] [Q&A]
What's a good way of skipping a member when serializing a response object?

Say I've got a class:
```
class Person:
    def __init__(self, name, age, secret):
        self.name = name
        self.age = age
        self.secret = secret
```
and I want to serialize an instance of `Person` as part of a response, but I don't want to include `secret`. 

A couple potential solutions I can think of:
1. Don't include `secret` in the response spec, but turn off strict validation
2. Deep copy the object, remove `secret`, then return
3. Modify `Person`'s `__format__` function to skip `secret`, which will eventually get called by `json.dumps`
4. Use a custom serializer that skips `secret` when it encounters an instance of `Person`

I don't like (1) because it requires turning of strict validation. (2) and (3) would work but are rather inelegant. I'm leaning most toward (4).

Any thoughts?

Thanks!

**Comment by chrisinmtown** [comment]
Just my opinion, it seems to me this is kind of a general OpenAPI spec question, not so much a /Connexion/ question, which might explain why nobody answered here.

Did you notice that Open API 3.0 defines a `writeOnly` property? The doc uses a password field as an example. I guess this would catch a situation where your code returns a field that it should not:
https://swagger.io/docs/specification/v3_0/data-models/data-types/#readonly-writeonly
